### PR TITLE
Add `AssumeRolePolicyDocument` statement conditions to `all_statements_condition` property

### PR DESCRIPTION
**Pull request by ocrawford555**
### Updates
- `all_statements_condition` property updated for `IAMRole` to include statements on `AssumeRolePolicyDocuments`.
- Update `CLOUDFORMATION_ACTIONS`.

### CODE REVIEW

File: pycfmodel/model/resources/iam_role.py [hunk 1]
```diff
@@ -51,3 +52,12 @@ def policy_documents(self) -> List[OptionallyNamedPolicyDocument]:
...
             result.append(OptionallyNamedPolicyDocument(name=policy.PolicyName, policy_document=policy.PolicyDocument))
         return result
+
+    @property
+    def all_statement_conditions(self) -> List[StatementCondition]:
+        assume_role_policy_document_conditions = [
+            statement.Condition
+            for statement in self.Properties.AssumeRolePolicyDocument.statement_as_list()
+            if statement.Condition
+        ]
```
Comment (w0rmr1d3r) [inline pycfmodel/model/resources/iam_role.py:62]: just thinking that this should be a property/method itself
while this property (`all_statement_conditions`) keep the return as it is, but returning both properties as one
what do you think?

File: tests/resources/test_iam_role.py [hunk 2]
```diff
@@ -39,6 +61,10 @@ def test_policies(iam_role):
     assert policies[0].PolicyName == "root"
 
 
+def test_all_conditions(iam_role):
+    assert len(iam_role.all_statement_conditions) == 2
```
Comment (w0rmr1d3r) [inline tests/resources/test_iam_role.py:65]: I'd add here more asserts, maybe check that the conditions returned are the same as the ones in the template?
Do you agree?

**Review by w0rmr1d3r — COMMENTED** [review]
Looking good!!
Some small suggestions, totally up to discussion.

File: pycfmodel/model/resources/iam_role.py (hunk 1 above)
Comment (ocrawford555) [reply pycfmodel/model/resources/iam_role.py]: See 61b14e8

**Review by oscarbc96 — APPROVED** [review]
🚀

**Review by w0rmr1d3r — APPROVED** [review]

**Pull request merged**